docs: Update website to include SIGHUP, SIGTERM, and SIGINT behaviour (#3140)
hashicorp/boundary: website/content/docs/configuration/worker/index.mdx

  `initial_upstreams`. This is currently only valid for workers using the PKI
  registration method and for workers directly connected to HCP Boundary.

## Signals

> On `SIGHUP`, a worker will reload its configuration file to pick up updated values for `initial_upstreams` and `tags`.
> All other updated values will be ignored.

Reviewer: ```suggestion
The `SIGHUP` signal causes a worker to reload its configuration file to pick up any updates for the `initial_upstreams` and `tags` values.
Any other updated values are ignored.
```

> `SIGTERM` or `SIGINT` will initiate graceful shutdown on a worker. The worker will wait for sessions to drain
> before shutting down. Workers in a graceful shutdown state will not receive new work, including session proxying, from the control plane.

Reviewer: ```suggestion
The `SIGTERM` and `SIGINT` signals initiate a graceful shutdown on a worker. The worker waits for any sessions to drain
before shutting down. Workers in a graceful shutdown state do not receive any new work, including session proxying, from the control plane.
```
